### PR TITLE
Restructure login

### PR DESCRIPTION
**Pull request by andrewseamanco**
- Restructured login so that google login and registration are on the same page
- Changed user and user js to get only take in username (and not last and first name)

### CODE REVIEW

File: bait-files/src/main/java/com/google/sps/servlets/LoginServlet.java [hunk 1]
```diff
@@ -35,11 +35,9 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
...
     String username = request.getParameter(USERNAME_PARAMETER);
-    String firstName = request.getParameter(FIRST_NAME_PARAMETER);
-    String lastName = request.getParameter(LAST_NAME_PARAMETER);
     Permission userPermission = Permission.USER;
 
-    User newUser = new User(userId, username, firstName, lastName, userPermission);
+    User newUser = new User(userId, username, userPermission);
     ObjectifyService.ofy().save().entities(newUser).now();
 
     response.sendRedirect("/profile.jsp");
```
Comment (imanherzi) [inline bait-files/src/main/java/com/google/sps/servlets/LoginServlet.java:43]: This is /history.jsp in the master branch now.

File: bait-files/src/main/java/com/google/sps/servlets/UserHistoryServlet.java [hunk 2]
```diff
@@ -64,8 +51,8 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     List<Review> reviews = allRequests.stream()
                                .filter(rev -> rev.userId.equals(userId))
                                .filter(rev -> rev.status == statusType)
```
Comment (imanherzi) [inline bait-files/src/main/java/com/google/sps/servlets/UserHistoryServlet.java:53]: Have you pulled from the master branch? There were a lot of changes to this servlet that were merged. This seems to be from before this was refactored.

File: bait-files/src/main/java/com/google/LoginFilter.java [hunk 3]
```diff
@@ -41,30 +41,15 @@ public void doFilter(ServletRequest req, ServletResponse res, FilterChain chain)
...
-      }
-      // Case: User is not registered and is trying to access restricted material
-      else {
-        RequestDispatcher requestDispatcher = request.getRequestDispatcher("WEB-INF/register.jsp");
-        requestDispatcher.forward(request, response);
-        return;
-      }
     } else {
       // Case: User is logged in
       if (request.getRequestURI().endsWith("profile.jsp")
```
Comment (imanherzi) [inline bait-files/src/main/java/com/google/LoginFilter.java:55]: File name has changed to history.jsp

File: bait-files/src/main/java/com/google/sps/servlets/LoginServlet.java (hunk 1 above)
Comment (andrewseamanco) [reply bait-files/src/main/java/com/google/sps/servlets/LoginServlet.java:43]: Renamed to history

File: bait-files/src/main/java/com/google/sps/servlets/UserHistoryServlet.java (hunk 2 above)
Comment (andrewseamanco) [reply bait-files/src/main/java/com/google/sps/servlets/UserHistoryServlet.java:53]: Just pulled

File: bait-files/src/main/java/com/google/LoginFilter.java (hunk 3 above)
Comment (andrewseamanco) [reply bait-files/src/main/java/com/google/LoginFilter.java:55]: It has been changed by pull

**Pull request merged**